src/viur/core/bones/uri.py

```diff
@@ -0,0 +1,204 @@
+import fnmatch
+import sys
+
+import logging
+from . import BaseBone, ReadFromClientError, ReadFromClientErrorSeverity
+from urllib.parse import urlparse, urlunparse
+import typing as t
+from collections.abc import Iterable
+from collections import namedtuple
+
+PORT_MIN: t.Final[int] = 1
+PORT_MAX: t.Final[int] = 2 ** 16 - 1
+
+
+class UriBone(BaseBone):
+    type = "uri"
+
+    def __init__(
+        self,
+        *,
+        accepted_protocols: str | t.Iterable[str] | None = None,
+        accepted_ports: int | str | t.Iterable[int] | t.Iterable[str] | None = None,
+        clean_get_params: bool = False,
+        domain_allowed_list: t.Iterable[str] | None = None,
+        domain_disallowed_list: t.Iterable[str] | None = None,
+        local_path_allowed: bool = False,
+        **kwargs
+    ):
+        """
+        The UriBone is used for storing URI and URL.
+
+        :param accepted_protocols: The accepted protocols can be set to allow only the provide protocols.
+        :param accepted_ports The accepted ports can be set to allow only the provide ports.
+        ..  code-block:: python
+            # Example
+            UriBone(accepted_ports=1)
+            UriBone(accepted_ports="2")
+            UriBone(accepted_ports="1-4")
+            UriBone(accepted_ports=(1,"2","4-10"))
+        :param clean_get_params: When set to True, the GET-parameter for the URL will be cleaned.
+        :param domain_allowed_list: If set, only the URLs that are matched with an entry of this iterable
+            will be accepted.
+        :param domain_disallowed_list: If set, only the URLs that are not matched
+            with an entry of this iterable will be accepted.
+        :param local_path_allowed: If True, the URLs that are local paths will be prefixed with "/".
+        """
+        super().__init__(**kwargs)
+        self.accepted_ports = set(sorted(UriBone._build_accepted_ports(accepted_ports), key=lambda rng: rng.start))
+
+        if range(PORT_MIN, PORT_MAX + 1) in self.accepted_ports:
+            self.accepted_ports = None  # all allowed
```

Comment:
Fix for ValueError, see below.

```suggestion
        if accepted_ports:
            self.accepted_ports = set(sorted(UriBone._build_accepted_ports(accepted_ports), key=lambda rng: rng.start))

            if range(PORT_MIN, PORT_MAX + 1) in self.accepted_ports:
                self.accepted_ports = None  # all allowed
        else:
            self.accepted_ports = None
```